types/src/front/lib/assistant.ts

```diff
@@ -241,8 +241,9 @@ const ANTHROPIC_DELIMITERS_CONFIGURATION = {
 const ANTHROPIC_TOOL_USE_META_PROMPT = `<tools_instructions>
 When using tools to answer the user's question, the assistant should follow these guidelines:
 
-1. Immediately before invoking a tool, think for one sentence in <thinking> tags about how it evaluates against the criteria for a good and bad tool use. Never emit any text beyond this thinking sentence before using the tool.
+1. Immediately before invoking a tool, think for one sentence in <thinking> tags about how it evaluates against the criteria for a good and bad tool use. The <thinking> section must never be longer than a single sentence. It is forbidden to emit any text outside of <thinking> tags before using a tool.
```

Comment:
We're prompting it to actually do one CoT always? Is that really what we want?

Can we avoid "It is forbidden" And rather use something more passive such as "Text should never be emitte outside of ..."

It is implies a lot of things: there is a system of rules and the assistant is forbidden some so it has agency so it has to decide of some things etc...

Reply:
> We're prompting it to actually do one CoT always? Is that really what we want?

Yes because this greatly limits the likelihood of having tokens not wrapped in `<thinking>` be emitted before the tool use (which almost always happens otherwise and looks quite ugly). It also shortens the amount of thinking from what I have seen.

All in all, it feels like a pretty good experience like this (feel free to try `@dust` on our workspace).

> Can we avoid "It is forbidden" And rather use something more passive such as "Text should never be emitte outside of ..."

I have tried several prompts and this is the one that worked consistently the best. But happy to make the suggested change 👍

Reply:
Bullish on avoiding implying there are rules that the model has to follow. This may come back as extraneous thoughts or response IMHO